graylog2-web-interface/src/views/components/sidebar/NavItem.tsx

```diff
@@ -21,27 +21,30 @@ import styled, { css } from 'styled-components';
 import Icon from 'components/common/Icon';
 import type { IconName } from 'components/common/Icon';
 
-type Props = {
+export type NavItemProps = {
   isSelected: boolean,
   title: string,
   icon: IconName,
   onClick: () => void,
-  showTitleOnHover: boolean,
+  showTitleOnHover?: boolean,
   sidebarIsPinned: boolean,
+  disabled?: boolean,
 };
 
 type ContainerProps = {
   isSelected: boolean,
   sidebarIsPinned: boolean,
+  disabled: boolean,
 };
 
-const Container = styled.div<ContainerProps>(({ theme: { colors, fonts }, isSelected, sidebarIsPinned }) => css`
+const Container = styled.div<ContainerProps>(({ theme: { colors, fonts }, isSelected, sidebarIsPinned, disabled }) => css`
   position: relative;
   z-index: 4; /* to render over SidebarNav::before */
   width: 100%;
   height: 40px;
   text-align: center;
-  cursor: pointer;
+  cursor: ${disabled ? 'not-allowed' : 'pointer'};
+  opacity: ${disabled ? 0.65 : 1};
```

Comment:
Maybe we can apply the opacity only for the icon and not for the label?
![image](https://github.com/Graylog2/graylog2-server/assets/46300478/d635c5df-3990-49ec-991c-9d0a69fcb290)